packages/components/src/components/vertical-tab/vertical-tab-headers.jsx

```diff
@@ -3,19 +3,52 @@ import VerticalTabWrapper from 'Components/vertical-tab/vertical-tab-wrapper.jsx
 import VerticalTabHeader from 'Components/vertical-tab/vertical-tab-header.jsx';
 import VerticalTabHeaderTitle from 'Components/vertical-tab/vertical-tab-header-title.jsx';
 
-const VerticalTabHeaders = props => (
-    <VerticalTabWrapper className={props.className}>
-        {props.header_title && <VerticalTabHeaderTitle header_title={props.header_title} />}
-        {props.items.map((item, idx) => (
-            <VerticalTabHeader
-                item={item}
-                onChange={props.onChange || undefined}
-                is_routed={props.is_routed}
-                selected={props.selected}
-                key={idx}
-            />
-        ))}
-    </VerticalTabWrapper>
-);
+class VerticalTabHeaders extends React.PureComponent {
+    ref = React.createRef();
+    headers = [];
+    state = { top: 0 };
+
+    componentDidMount() {
+        this.headers = this.ref.current.querySelectorAll('.dc-vertical-tab__header__link');
+        this.setState({ top: this.offset_top });
+    }
+
+    componentDidUpdate() {
+        this.setState({ top: this.offset_top });
+    }
+
+    get offset_top() {
+        const { selected } = this.props;
+
+        const selected_el = [...this.headers].filter(
```

Comment:
`.find`? do we need to spread out the headers into a new array first?

Reply:
yes, we need to use spread because `this.headers` are NodeList or HTMLCollection object, so we need to convert those into javascript array